src/main.rs

```diff
@@ -0,0 +1,187 @@
+extern crate rustc_serialize;
+extern crate docopt;
+#[macro_use]
+extern crate lazy_static;
+extern crate regex;
+extern crate tabwriter;
+
+mod benchmark;
+mod utils;
+
+use docopt::Docopt;
+use regex::Regex;
+use tabwriter::TabWriter;
+
+use benchmark::{Benchmark, parse_benchmarks};
+use utils::find_overlap;
+
+use std::io;
+use std::io::prelude::*;
+use std::fs::File;
+
+const USAGE: &'static str = r#"
+Compare Rust micro-benchmarks by saving the output of the benchmark to file
+and providing it into this command.
+The first version takes two file and compares the common bench-tests.
+The second version takes two names of implementations and one or more files,
+and compares the common bench-tests of the two implementations based on the
+name scheme "implementation::test".
+
+Usage: cargo benchcmp [options] <file> <file>
+       cargo benchcmp [options] <name> <name> [--] <file>...
+       cargo benchcmp -h | --help
+
+
+Options:
+  -h, --help            show this help message and exit
+  --threshold <n>       only show comparisons with a percentage change greater
+                        than this threshold
+  --variance            show variance
+  --show <option>       show regressions, improvements or both [default: both]
+  --strip-fst <regex>   a regex to strip from first benchmarks' names
+  --strip-snd <regex>   a regex to strip from second benchmarks' names
+"#;
+
+#[derive(Debug, RustcDecodable)]
+struct Args {
+    flag_threshold: Option<u8>,
+    flag_variance: bool,
+    flag_show: ShowOption,
+    flag_strip_fst: Option<String>,
+    flag_strip_snd: Option<String>,
+    arg_name: Option<[String; 2]>,
+    arg_file: Vec<String>,
+}
+
+#[derive(Debug, RustcDecodable, PartialEq, Eq)]
+enum ShowOption {
+    Regressions,
+    Improvements,
+    Both,
+}
+
+macro_rules! err_println {
+    ($fmt:expr) => (err_print!(concat!($fmt, "\n")));
+    ($fmt:expr, $($arg:tt)*) => (err_print!(concat!($fmt, "\n"), $($arg)*));
+}
+
+macro_rules! err_print {
+    ($($arg:tt)*) => (io::stderr().write_fmt(format_args!($($arg)*)).unwrap(););
+}
+
+macro_rules! create_replace_fn {
+    ($e:expr, $s:expr) => { match $e {
+        None => Box::new(|s: &str| String::from(s)),
+        Some(s) => {
+            match Regex::new(s.as_str()) {
+                Ok(re) => Box::new(move |s: &str| re.replace(s, "")),
+                Err(e) => {
+                    err_println!(concat!("ERROR: strip_", $s, ": {}"), e);
+                    return;
+                }
+            }
+        }
+    }}
+}
+
+// The create_replace_fn macro should use Box::new(String::from), but the compiler's type inference
+//  can't handle it. 
+#[allow(redundant_closure)]
+fn main() {
```

Comment:
I think there are quite a few `unwrap` calls in this main function that will panic in unremarkable circumstances. For example, if the user passes a file path that doesn't exist or is otherwise not readable, then this program will panic and produce an ugly error message.

I'd recommend factoring out much of the `main` function into a different function that returns a `Result<(), Box<Error+Send+Sync>>`. Then most of your `x.unwrap()` calls can be replaced with `try!(x)`. `main` itself can then be reduced to the `Docopt` call and a match on the returned `Result` that prints a nice message to `stderr` if there was an error.